src/groups/semidirect_product_group.jl

```diff
@@ -123,13 +123,15 @@ function _compose!(G::SemidirectProductGroup, x, p, q)
     M = base_manifold(G)
     N, H = M.manifolds
     A = G.op.action
+    x_tmp = allocate(x)
     np, hp = submanifold_components(G, p)
     nq, hq = submanifold_components(G, q)
-    nx, hx = submanifold_components(G, x)
+    nx, hx = submanifold_components(G, x_tmp)
     compose!(H, hx, hp, hq)
     nxtmp = apply(A, hp, nq)
     compose!(N, nx, np, nxtmp)
     @inbounds _padpoint!(G, x)
+    copyto!(x, x_tmp)
```

Comment:
This is the bugfix from https://github.com/JuliaManifolds/Manifolds.jl/issues/609 -- I just didn't want to open yet another PR...

Reply:
Basically `p` and `x` were the same variable and overwriting `x` caused `p` to change.

Reply:
Oh, indeed that was a side effect that should not happen, well-spotted. Sure we can add it here as well. Though at some point we should maybe also start a Changelog here (to see where we did what also here).

Reply:
Yes, I would really like to have a changelog here but I'm usually less enthusiastic about actually writing one :sweat_smile:

Reply:
Then let's start one and I take the role (similar to the docs) of reminding everyone (hopefully, forgot that twice in Manopt already as well and did one PR with 3 changelog entries).

I think it's worth it, especially for the really condensed overview of when which topic/feature appeared.